Client/Frontend/Widgets/AutocompleteTextField.swift

```diff
@@ -39,9 +40,11 @@ class AutocompleteTextField: UITextField, UITextFieldDelegate {
 
     private func commonInit() {
         super.delegate = self
+        super.addTarget(self, action: "SELtextDidChange:", forControlEvents: UIControlEvents.EditingChanged)
         notifyTextChanged = debounce(0.1, {
             if self.editing {
-                self.autocompleteDelegate?.autocompleteTextField(self, didEnterText: self.text)
+                let enteredText = self.text.substringToIndex(advance(self.text.startIndex, self.enteredTextLength, self.text.endIndex))
+                self.autocompleteDelegate?.autocompleteTextField(self, didEnterText: enteredText)
```

Comment:
OK. I see we are doing this `enteredText` calculation in lots of places. Maybe it would make sense to just store `enteredText` directly as a class variable, and remove `enteredTextLength`? If there are places we still need the entered text length, we could just use `count(enteredText)`. And then we don't need to use `substringToIndex` everywhere.